services/system/Accounts/plugin/src/lib.rs

```diff
@@ -29,6 +43,39 @@ impl Accounts for AccountsPlugin {
         Err(NotYetImplemented.err("add_auth_service"))
     }
 
+    fn get_account(name: String) -> Result<Option<AccountTypes::Account>, CommonTypes::Error> {
+        let query = format!(
+            "query {{ getAccount(account: \"{}\") {{ accountNum, authService, resourceBalance }} }}",
+            AccountNumber::from(name.as_str())
+        );
+
+        let account_result = Server::post_graphql_get_json(&query)
+            .map_err(|e| QueryError.err(&e.message))
+            .and_then(|result| {
+                serde_json::from_str(&result).map_err(|e| QueryError.err(&e.to_string()))
+            });
+
+        let account = account_result.and_then(|response_root: ResponseRoot| {
+            response_root
+                .data
+                .getAccount
+                .ok_or_else(|| InvalidAccountNumber.err(&name))
+        })?;
+
+        if account.accountNum.value != AccountNumber::from_exact(&name).unwrap().value {
```

Comment:
Don't need the roundtrip check. Only names that roundtrip could ever be returned from this query.

Reply:
Not worth having the plugin do the work that will save node CPU / Account service time?

Reply:
If that's the goal then it should be done before the query is made. At this point the query is complete so it's too late to check if it was a valid account name.
Also, if you keep it, you should change the error to say, 'invalid account *name*' rather than `number`.